platform/pulpcore/app/upload.py

```diff
@@ -0,0 +1,34 @@
+import hashlib
+from django.core.files.uploadhandler import TemporaryFileUploadHandler
+from django.core.files.uploadedfile import TemporaryUploadedFile
+
+
+class PulpTemporaryUploadedFile(TemporaryUploadedFile):
+    """
+    A file uploaded to a temporary location in Pulp.
+    """
+
+    def __init__(self, name, content_type, size, charset, content_type_extra=None):
+        self.hashers = {}
+        for hasher in hashlib.algorithms_guaranteed:
+            self.hashers[hasher] = getattr(hashlib, hasher)()
+        super().__init__(name, content_type, size, charset, content_type_extra)
+
+
+class HashingFileUploadHandler(TemporaryFileUploadHandler):
+    """
+    Upload handler that streams data into a temporary file.
+    """
+    def new_file(self, file_name, *args, **kwargs):
+        """
+        Create the file object to append to as data is coming in.
+        """
+        super().new_file(file_name, *args, **kwargs)
```

Comment:
This will create a file whose reference I think will be overwritten by the next line. I also think we don't need or want that file at all. Is there anything that calling the superclass ``new_file()`` gets us that we need? Or could that be left out?

https://docs.djangoproject.com/en/1.11/_modules/django/core/files/uploadhandler/#TemporaryFileUploadHandler